lib/ohloh_scm/git_scm.rb

```diff
@@ -2,5 +2,63 @@
 
 module OhlohScm
   class GitScm < Scm
+    def initialize(base:, url:, branch_name:, username:, password:)
+      super
+      @branch_name = branch_name || 'master'
+    end
+
+    # == Example:
+    #   remote_base = OhlohScm::Factory.get_base(url: 'https://github.com/ruby/ruby')
+    #   local_base = OhlohScm::Factory.get_base(url: '/tmp/ruby-src')
+    #   local_base.scm.pull(remote_base.scm)
+    def pull(from)
+      clone_or_fetch(from)
+    end
+
+    private
+
+    def clone_or_fetch(remote_scm)
+      if status.branch?(branch_name)
```

Comment:
How are we handling the call back method to the client?  Is this a method that will be in the from object?

Reply:
Fair point.
For git clone there are only 2 steps. So I then considered not having an update progress for the Fetch. I now see that GitSvn/Cvs pull implementations will need a callback since they have more steps on account of commit conversions. I was hoping to peek at the other implementations and go accordingly.

Reply:
Added the callback object for update progress.

Reply:
Maybe I misunderstood, but I though PDP had wanted a method on the from object to have an update method.  I think the from object is a local_scm(?)